providers/openfeature-go-feature-flag-provider/lib/openfeature/go-feature-flag/error/errors.rb

```diff
@@ -0,0 +1,76 @@
+# Define a custom error class
+require "open_feature/sdk/provider/error_code"
+
+module OpenFeature
+  module GoFeatureFlag
+    class FlagNotFoundError < StandardError
+      attr_reader :response, :error_code, :error_message
+
+      def initialize(response, flag_key)
+        error_message = "Flag not found: #{flag_key}"
+        super(error_message)
+        @response = response
+        @error_code = OpenFeature::SDK::Provider::ErrorCode::FLAG_NOT_FOUND
```

Comment:
nitpick: Since you're inside of the `OpenFeature` module, you can drop it from the module resolution here to slightly less verbose.

```suggestion
        @error_code = SDK::Provider::ErrorCode::FLAG_NOT_FOUND
```